nomos-services/mix/src/lib.rs

```diff
@@ -77,73 +79,71 @@ where
 
     async fn run(mut self) -> Result<(), overwatch_rs::DynError> {
         let Self {
-            mut service_state,
+            service_state,
             mut backend,
             network_relay,
         } = self;
         let mix_config = service_state.settings_reader.get_updated_settings();
-
+        let cryptographic_processor =
+            CryptographicProcessor::new(mix_config.message_blend.cryptographic_processor);
         let network_relay = network_relay.connect().await?;
         let network_adapter = Network::new(network_relay);
 
-        // Spawn Persistent Transmission
-        let (transmission_schedule_sender, transmission_schedule_receiver) =
-            mpsc::unbounded_channel();
-        let (emission_sender, mut emission_receiver) = mpsc::unbounded_channel();
-        tokio::spawn(async move {
-            persistent_transmission(
-                mix_config.persistent_transmission,
-                transmission_schedule_receiver,
-                emission_sender,
-            )
-            .await;
-        });
-
-        // Spawn Message Blend and connect it to Persistent Transmission
-        let (new_message_sender, new_message_receiver) = mpsc::unbounded_channel();
-        let (processor_inbound_sender, processor_inbound_receiver) = mpsc::unbounded_channel();
-        let (fully_unwrapped_message_sender, mut fully_unwrapped_message_receiver) =
-            mpsc::unbounded_channel();
-        tokio::spawn(async move {
-            MessageBlend::new(
-                mix_config.message_blend,
-                new_message_receiver,
-                processor_inbound_receiver,
-                // Connect the outputs of Message Blend to Persistent Transmission
-                transmission_schedule_sender,
-                fully_unwrapped_message_sender,
-            )
-            .run()
-            .await;
-        });
-
-        // A channel to listen to messages received from the [`MixBackend`]
-        let mut incoming_message_stream = backend.listen_to_incoming_messages();
+        // tier 1 persistent transmission
+        let (persistent_sender, persistent_receiver) = mpsc::unbounded_channel();
+        let mut persistent_transmission_messages =
+            UnboundedReceiverStream::new(persistent_receiver)
+                .persistent_transmission(mix_config.persistent_transmission);
+
+        // tier 2 blend
+        let mut blend_messages = backend
+            .listen_to_incoming_messages()
+            .blend(mix_config.message_blend);
+
+        // local messages, are bypassed and send immediately
+        let mut local_messages = service_state
+            .inbound_relay
+            .map(|ServiceMessage::Mix(message)| {
+                wire::serialize(&message)
+                    .expect("Message from internal services should not fail to serialize")
+            });
```

Comment:
Here is the sauce. we got:
* The persistent tier 1
* The blend tier 2
* The local messages that are now encoded and handled straight forward